pinot-core/src/main/java/org/apache/pinot/core/plan/GroupByPlanNode.java

```diff
@@ -78,7 +78,8 @@ private GroupByOperator buildNonFilteredGroupByPlan() {
 
     // Use star-tree to solve the query if possible
     List<StarTreeV2> starTrees = _indexSegment.getStarTrees();
-    if (starTrees != null && !_queryContext.isSkipStarTree()) {
+    if (!_queryContext.isNullHandlingEnabled() && !filterOperator.isResultEmpty() && starTrees != null
```

Comment:
so we were missing null handling enable flag before as well here? :-P

Reply:
Unfortunately yes..